src/me.rs

```diff
@@ -86,35 +86,40 @@ pub fn get_subset_predictors<T: Pixel>(
 ) -> ArrayVec<[MotionVector; 17]> {
   let mut predictors = ArrayVec::<[_; 17]>::new();
 
-  // Zero motion vector
+  // Add a candidate predictor, aligning to fullpel and filtering out zero mvs.
+  let add_cand = |predictors: &mut ArrayVec<[MotionVector; 17]>,
+                  cand_mv: MotionVector| {
+    let cand_mv = cand_mv.quantize_to_fullpel();
+    if !cand_mv.is_zero() {
+      predictors.push(cand_mv)
+    }
+  };
+
+  // Zero motion vector, don't use add_cand since it skips zero vectors.
```

Comment:
Why not make add_cand push the MV without the `quantize_to_fullpel()` call? That seems more straightforward than making it a special case at this level.